Content.Shared/_NF/Trade/TradeCrateComponent.cs

```diff
@@ -1,9 +1,58 @@
-namespace Content.Shared._NF.Trade.Components;
+using System.Threading;
+using Content.Shared.Cargo;
+using Robust.Shared.GameStates;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom;
+
+namespace Content.Shared._NF.Trade;
 
 /// <summary>
-/// This is used to mark an entity to be used in a trade crates
+/// This is used to mark an entity to be used as a trade crate
 /// </summary>
-[RegisterComponent]
+[RegisterComponent, NetworkedComponent, AutoGenerateComponentState, AutoGenerateComponentPause, Access(typeof(SharedCargoSystem))]
 public sealed partial class TradeCrateComponent : Component
 {
+    /// <summary>
+    /// The value of the crate, in spesos, when at its destination.
+    /// </summary>
+    [DataField(serverOnly: true)]
+    public int ValueAtDestination;
+
+    /// <summary>
+    /// The value of the crate, in spesos, when delivered elsewhere.
+    /// </summary>
+    [DataField(serverOnly: true)]
+    public int ValueElsewhere;
+
+    /// <summary>
+    /// If non-zero, this crate will be an express delivery.
+    /// </summary>
+    [DataField(serverOnly: true, customTypeSerializer: typeof(TimeOffsetSerializer))]
+    public TimeSpan ExpressDeliveryDuration = TimeSpan.Zero;
+
+    /// <summary>
+    /// If non-null, the package must be redeemed before this time to arrive unpenalized.
+    /// </summary>
+    [ViewVariables, AutoNetworkedField, AutoPausedField]
```

Comment:
this needs a timeoffset serializer
doesnt need AutoNetworked because its not accessed on the client

Reply:
Why does it need the TimeOffsetSerializer?  It's not read in from YAML.  It's not a DataField.  It's defined based on the one that is, which has the serializer.  I talked about this before.

If the complaint is "that's dumb, don't do that", instead reusing one field, sure, there are , but make that claim.

"It's not accessed on the client": the examine could be moved to Shared, and that's the only client-side thing that this is useful for - could set its own tape to inactive as well, I suppose.

Reply:
TimeOffsetSerializer and AutoPausedField should be used for all TimeSpan that can change in runtime. The offset serializer is there to account for pause time and to prevent this from being saved to the map file.

It's not needed for static TimeSpan that never change, such as pre-defined delays, as these can be just DataFields with no special attributes.

We're never accessing component *state* on the client, just the component itself, so we do not need to network any of the fields.

Reply:
This was based off of the mail.  If you want to rewrite the mail system to use your own suggestions, great, I'll follow in your stead, but as-is, the mail system works, your suggestions aren't a complete solution, and I don't want to add an update call that regularly queries every trade crate in existence instead of using a timer.